pkgs/development/compilers/vlang/default.nix

```
@@ -76,12 +96,32 @@ stdenv.mkDerivation rec {
     cp $HOME/vtest.v $out/lib/cmd/tools/vtest.v
   '';
 
-  meta = with lib; {
-    homepage = "https://vlang.io/";
-    description = "Simple, fast, safe, compiled language for developing maintainable software";
-    license = licenses.mit;
-    maintainers = with maintainers; [ Madouura ];
-    mainProgram = "v";
-    platforms = platforms.all;
-  };
+  preFixup = ''
+    # We need to patch the shebang of file "lib/cmd/tools/git_pre_commit_hook.vsh"
+    # and file "lib/vlib/v/tests/script_with_no_extension"
+    # Going from "#!/usr/bin/env -S v -raw-vsh-tmp-prefix tmp"
+    # To "#!$out/v -raw-vsh-tmp-prefix tmp"
```

```suggestion
    # We need to patch the shebangs to point to our v
```

we don't need to repeat the literal code. Also can this be done via patchShebangs?

---

The default shebang patcher didn't recognize those shebangs (probably due to the `#!/usr/bin/env -S`, so I went this route instead. While it would probably work if patched with `patchShebangs`, the real reason the V developers decided to use `#!/usr/bin/env` is that V could be installed from anywhere. It made more sense to me to just fix it to our actual V instead

---

Change suggestion fixed in https://github.com/COLAMAroro/nixpkgs/blob/dcaed354bd52104677eb30460088e627ce74e465/pkgs/development/compilers/vlang/default.nix#L118